### PR TITLE
Adds a synchronous queue for SPI transfers

### PR DESCRIPTION
**Pull request by johnnyman727**

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -228,7 +230,21 @@ Attiny.prototype.setIRQCallback = function(callback) {
...
 Attiny.prototype.transceive = function(dataBuffer, callback) {
-  this.spi.transfer(dataBuffer, callback);
+  var self = this;
+  // Put this new transfer into the queue
+  this.messageQueue.place(function() {
+    console.log('send');
+    // Activate the transfer when it's first in the queue
+    self.spi.transfer(dataBuffer, function(err, data) {
+      console.log('done!');
+      // Return data or err as necessary
```
Comment (rwaldron) [inline lib/index.js]: Should these console.log's still be here, or is this still WIP?

Comment (johnnyman727) [inline lib/index.js]: WIP - just edited the PR title to match.

**Comment by draco2003** [comment]
Any updates on this?

**Comment by johnnyman727** [comment]
@draco2003 I was able to confirm that this PR does improve behavior a bit but it does not solve all the problems we've seen. I haven't had time to investigate further due to increased workload outside of the project.

**Comment by johnnyman727** [comment]
I'm going to go ahead and merge this since it seems to decrease the frequency of some issues.

**Pull request merged**